### ISSUE TITLE
Add capability to change associated infrastructure mapping when editing a migration plan

### ISSUE THREAD

**Issue by mturley** [ManageIQ/manageiq-v2v#638]
To be completed after #620 is merged. Currently we do not allow the user to change infrastructure mappings when editing a migration plan, but doing so should invalidate the rest of the wizard and prevent pre-filling the steps (other than the General step) with any data from the plan being edited.

This also affects orphaned plans (plans with no associated mapping), which are currently not allowed to be edited. They should be editable too, and allow the user to select a non-orphaned migration plan.

**Comment by michaelkro** [comment]
I think we can close this issue out since it seems that we have general agreement that it makes more sense to delete and create a new plan if the mapping needs to be changed

**Comment by mturley** [comment]
Sounds good to me.
